components/Datasets/FileEditModal.vue

```diff
@@ -144,9 +234,73 @@ const route = useRoute()
 
 const resourceForm = ref(cloneDeep(props.resource))
 const open = ref(false)
+const hasFileChanged = ref(false)
 
 onMounted(() => {
   if (props.openOnMounted) open.value = true
+  
+  // Debug: Log current resource file information
+  console.log('FileEditModal - Current resource file info:', {
+    resourceForm: resourceForm.value,
+    originalResource: props.resource.resource,
+    filetype: resourceForm.value.filetype,
+    hasFile: !!resourceForm.value.file,
+    fileInfo: resourceForm.value.file,
+    checksum: props.resource.resource?.checksum,
+    filesize: props.resource.resource?.filesize,
+    title: props.resource.resource?.title
+  })
+  
+  // Debug: Log the complete originalResource structure
+  console.log('FileEditModal - Complete originalResource:', props.resource.resource)
+  console.log('FileEditModal - originalResource JSON:', JSON.stringify(props.resource.resource, null, 2))
+  
+  // Function to check if file has changed based on metadata
+  const checkFileChange = () => {
```

Comment:
I think this is better ! Somehow, when I change a file, it never goes to `checkFileChange`. Is it working on your side ?
Also it's not checking for changes for a remote resource.